route/build/src/kwt_route.f90

```diff
@@ -408,216 +409,218 @@ SUBROUTINE QROUTE_RCH(IENS,JRCH,    & ! input: array indices
+    KROUTE(IENS,JRCH)%KWAVE(0:NR)%QF=Q_JRCH(0:NR); KROUTE(IENS,JRCH)%KWAVE(NR+2:NQ2+1)%QF=Q_JRCH(NR+1:NQ2)
+    KROUTE(IENS,JRCH)%KWAVE(0:NR)%TI=TENTRY(0:NR); KROUTE(IENS,JRCH)%KWAVE(NR+2:NQ2+1)%TI=TENTRY(NR+1:NQ2)
+    KROUTE(IENS,JRCH)%KWAVE(0:NR)%TR=T_EXIT(0:NR); KROUTE(IENS,JRCH)%KWAVE(NR+2:NQ2+1)%TR=T_EXIT(NR+1:NQ2)
+    KROUTE(IENS,JRCH)%KWAVE(0:NR)%RF=FROUTE(0:NR); KROUTE(IENS,JRCH)%KWAVE(NR+2:NQ2+1)%RF=FROUTE(NR+1:NQ2)
+    KROUTE(IENS,JRCH)%KWAVE(0:NQ2+1)%QM=-9999
+    ! implement water use
+    !IF (NUSER.GT.0.AND.UCFFLAG.GE.1) THEN
+      !CALL EXTRACT_FROM_RCH(IENS,JRCH,NR,Q_JRCH,T_EXIT,T_END,TNEW)
+    !ENDIF
+    ! free up space for the next reach
+    deallocate(Q_JRCH,TENTRY,T_EXIT,FROUTE,STAT=IERR)   ! FROUTE defined in this sub-routine
+    if(ierr/=0)then; message=trim(message)//'problem deallocating space for [Q_JRCH, TENTRY, T_EXIT, FROUTE]'; return; endif
+    ! ***
+    ! remove flow particles from the most downstream reach
+    if(jRch==ixOutlet)then
+      if(jRch==ixPrint) print*, 'NETOPO(JRCH)%DREACHI = ', NETOPO(JRCH)%DREACHI
+      if(NETOPO(JRCH)%DREACHI > 0)then
+        message=trim(message)//'downstream reach index of outlet reach is greater than zero'
+        ierr=20; return
+      endif
+    endif
+    ! if the last reach or lake inlet (and lakes are enables), remove routed elements from memory
+    IF (NETOPO(JRCH)%DREACHI.LT.0 .OR. &  ! if the last reach, then there is no downstream reach
+        (LAKEFLAG.EQ.1.AND.NETOPO(JRCH)%LAKINLT)) THEN ! if lake inlet
+      ! copy data to a temporary wave
+      if (allocated(NEW_WAVE)) THEN
+        DEALLOCATE(NEW_WAVE,STAT=IERR)
+        if(ierr/=0)then; message=trim(message)//'problem deallocating space for NEW_WAVE'; return; endif
+      endif
+      ALLOCATE(NEW_WAVE(0:NN),STAT=IERR)  ! NN = number non-routed (the zero element is the last routed point)
+      if(ierr/=0)then; message=trim(message)//'problem allocating space for NEW_WAVE'; return; endif
+      NEW_WAVE(0:NN) = KROUTE(IENS,JRCH)%KWAVE(NR+1:NQ2+1)  ! +1 because of the interpolated point
+      ! re-size wave structure
+      if (allocated(KROUTE(IENS,JRCH)%KWAVE)) THEN
+        deallocate(KROUTE(IENS,JRCH)%KWAVE,STAT=IERR)
+        if(ierr/=0)then; message=trim(message)//'problem deallocating space for KROUTE'; return; endif
+      endif
+      allocate(KROUTE(IENS,JRCH)%KWAVE(0:NN),STAT=IERR)  ! again, the zero element for the last routed point
+      if(ierr/=0)then; message=trim(message)//'problem allocating space for KROUTE'; return; endif
+      ! copy data back to the wave structure and deallocate space for the temporary wave
+      KROUTE(IENS,JRCH)%KWAVE(0:NN) = NEW_WAVE(0:NN)
+      DEALLOCATE(NEW_WAVE,STAT=IERR)
+      if(ierr/=0)then; message=trim(message)//'problem deallocating space for NEW_WAVE'; return; endif
+    endif  ! (if JRCH is the last reach)
+    ! get size of wave number for a reach
+    return
```

Comment:
this seems odd to me. What is happening here?

Reply:
I think the comment can be removed (caused confusion?)